homeassistant/helpers/state.py

```diff
@@ -202,11 +203,12 @@ def state_as_number(state):
 
     Raises ValueError if this is not possible.
     """
-    if state.state in (STATE_ON, STATE_LOCKED, STATE_ABOVE_HORIZON,
-                       STATE_OPEN):
+    if state.state in (STATE_ON, STATE_LOCKED, STATE_ABOVE_HORIZON, STATE_OPEN,
+                       STATE_HOME, STATE_PLAYING):
```

Comment:
I agree, but we should take more devs opinion and it should go in another PR that targets that topic specifically. It will probably need some discussion on how to handle breakage. We've used migration scripts in the past eg for influxdb. You're welcome to start a PR or RFC issue for discussion.